app/multigraph_cli.cpp

```diff
@@ -132,6 +133,15 @@ void MultigraphCLI::execute_find_hamiltonian_extension() const {
         }
         std::cout << std::endl;
     }
+
```

Comment:
I think you should add to exact algorithm as well but use maxCyclesolver and get maxCycleVertices to get cycles.